Automattic/wp-calypso
# Make dialogs scrollable when their content is larger than the window height

This PR addresses some issues with the Dialog component. The main issue fixed is how modals that are larger than the current viewport are handled. Currently in these cases the edges of the modal bleed off-screen and are unaccessible. This can result in action buttons being un-clickable. 😱

I also removed the `color` declaration on `.dialog__content` so colors are now inherited as expected.

<hr />

To test: 

* Open up a dialog, for example the 'Add link' modal in the Editor. 
* Resize your browser window so that it is smaller than the Dialog.
* Notice the dialog content now scrolls and the action buttons are fixed in position.
* Check the media modal, (and any other modals) are working as expected.

<hr />

EDIT: Adding todo:

- [x] Ensure Dialogs are never larger than the browser window
- [x] When Dialog content is taller than browser window, make it scroll
- [x] Don't allow scrolling on the 'background content' when Dialogs are open
- [x] If the dialog has action buttons, fix them in position so they're always visible if/when the content scrolls
- [x] Remove 'color' declaration currently applied to all Dialog content
- [x] Ensure media Dialog in unaffected by any of these changes
